pipeline/matching/core/engine.py

```diff
@@ -26,7 +26,7 @@ class UserNotMatchedException(Exception):
 
 class MatchingEngine:
     def __init__(
-        self, name, generators, ranker, finalizer, return_internal_data=False
+        self, name, generators, ranker, finalizer, return_internal_data=True
```

Comment:
This is causing tests to fail, instead of changing the default value to True, we should only set it to True when we run the pipeline.

Change the default value here back to False.

Then, go to this line in the matching flow code: https://github.com/scarletstudio/butterfly/blob/main/pipeline/flows/matching.py#L47

And change it to have:

```python
engine: MatchingEngine = Engine(return_internal_data=True)
```

That will achieve your goal of turning on internal data during the pipeline run, but leaving it defaulted to False so that the test cases pass.

Reply:
I'm getting a TypeError after making the change you're suggesting here. However, if I change it to this instead:

engine: MatchingEngine(return_internal_data=True) = Engine()

Everything runs fine. For the time being, I'll go with this.